# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../dltrace/dltrace-0.5-deprecated-asm-headers.patch
# Copyright (C) 2011 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

--- dltrace-0.5/lib/libplump/include/plumpconfig.h.orig	2011-04-29 11:16:48.700000255 +0200
+++ dltrace-0.5/lib/libplump/include/plumpconfig.h	2011-04-29 11:17:56.572000254 +0200
@@ -19,7 +19,7 @@
 #define REGISTER_STRUCT_TYPE_REG              0x0002
 
 #if defined(OS_LINUX)
-	#include <asm/user.h>     // struct user_regs_struct
+	#include <sys/user.h>     // struct user_regs_struct
 
 	extern int ptrace (int request, int pid, ...);
 #elif defined(OS_FREEBSD)    // struct reg
--- dltrace-0.5/lib/libplump/src/core/registers.c.orig	2011-04-29 11:09:08.380000254 +0200
+++ dltrace-0.5/lib/libplump/src/core/registers.c	2011-04-29 11:09:59.008000254 +0200
@@ -99,17 +99,17 @@
 				else if (REGISTER_NAME(want[x].name) == REGISTER_NAME(REGISTER_IA32_ESI)) 
 					offset = &((struct user_regs_struct *)reg)->esi;
 				else if (REGISTER_NAME(want[x].name) == REGISTER_NAME(REGISTER_IA32_SS)) 
-					offset = &((struct user_regs_struct *)reg)->ss;
+					offset = &((struct user_regs_struct *)reg)->xss;
 				else if (REGISTER_NAME(want[x].name) == REGISTER_NAME(REGISTER_IA32_CS)) 
-					offset = &((struct user_regs_struct *)reg)->cs;
+					offset = &((struct user_regs_struct *)reg)->xcs;
 				else if (REGISTER_NAME(want[x].name) == REGISTER_NAME(REGISTER_IA32_DS)) 
-					offset = &((struct user_regs_struct *)reg)->ds;
+					offset = &((struct user_regs_struct *)reg)->xds;
 				else if (REGISTER_NAME(want[x].name) == REGISTER_NAME(REGISTER_IA32_ES)) 
-					offset = &((struct user_regs_struct *)reg)->es;
+					offset = &((struct user_regs_struct *)reg)->xes;
 				else if (REGISTER_NAME(want[x].name) == REGISTER_NAME(REGISTER_IA32_FS)) 
-					offset = &((struct user_regs_struct *)reg)->fs;
+					offset = &((struct user_regs_struct *)reg)->xfs;
 				else if (REGISTER_NAME(want[x].name) == REGISTER_NAME(REGISTER_IA32_GS)) 
-					offset = &((struct user_regs_struct *)reg)->gs;
+					offset = &((struct user_regs_struct *)reg)->xgs;
 				break;
 #elif defined(OS_FREEBSD)
 			case REGISTER_STRUCT_TYPE_REG:
--- dltrace-0.5/include/dltrace.h.orig	2011-04-29 11:15:17.348000255 +0200
+++ dltrace-0.5/include/dltrace.h	2011-04-29 11:15:57.048000255 +0200
@@ -22,7 +22,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <signal.h>
-#include <asm/page.h>
+#include <sys/user.h>		/* PAGE_SIZE */
 // asm/page.h -- linux 
 // i386/vmparam.h -- nbsd 
 // i386/param.h -- obsd 
